WordPress/gutenberg
# Remove `showFixedToolbar` from `useShowBlockTools`

## What?

Prevent unnecessary rerenders of post editor header caused by https://github.com/WordPress/gutenberg/pull/58979.

## Why?

Trying to improve "Selecting blocks"/focus editor performance.

## How?

Move the header `<BlockToolbar />` and elements that relied on `showFixedToolbar` to only run when the fixed toolbar is active. The post editor `<Header />` was rerendering every time a new block was selected because of this check.

## Testing Instructions




Post and site editors should work as normal with both popover toolbar and header toolbar.

Run Performance test of selecting blocks on both this branch and trunk. This branch should have a faster "focus" time.
`npm run test:performance -- -g "Selecting blocks"`

The `focus` metric is routinely 10-15% lower on this branch than on `trunk`.

Using the React Profiler, check the Header does not rerender when:
- On a post
- Floating Toolbar (NOT fixed toolbar)
- Have multiple empty paragraphs
- Select an empty paragraph
- Start Profiler
- Click a new empty paragraph
- Stop Profiler
- Check the Header does not rerender

### Testing Instructions for Keyboard


## Screenshots or screencast 

React Profiler for this branch (no expensive Header rerender)

https://github.com/WordPress/gutenberg/assets/967608/056e0865-5668-4bb4-b0bb-4be571394100

React Profiler for `trunk` with Header rerender

https://github.com/WordPress/gutenberg/assets/967608/0f9a0c25-7503-4f3b-84b1-73ea3842b82b
